### ISSUE TITLE
FR: `jj resolve` should affirmatively report whether conflicts in the parent were resolved

### ISSUE THREAD

**Issue by davidbarsky** [jj-vcs/jj#5188]
**Is your feature request related to a problem? Please describe.**
When using `jj resolve`, I'd find it to be extremely helpful if `jj resolve` affirmatively reported whether all conflicts in the parent commit were resolved. Today, it only reports whether there are any [_remaining_ conflicts](https://github.com/jj-vcs/jj/blob/e5361c6cc011f5301dad8e0e9461f27bcc0863a3/cli/src/commands/resolve.rs#L121-L141), which leads me to second-guess myself and run `jj st`.

**Describe the solution you'd like**
`jj resolve` should print additional context about the working copy's conflict resolution state. For example, if a `jj resolve` command completes successfully and no more conflicts remain, `jj resolve` should write "Conflict in parent commit has been resolved in working copy" as `jj status` does.

**Describe alternatives you've considered**
- I currently run run `jj st` after a `jj resolve`, which works, but it's an additional command.
- I could probably setup my shell prompt to report on any remaining conflicts, but I'm lazy and forgot to do a similar integration with Sapling. I'll probably do this after I'm done writing this issue.
- A richer GUI integration with jj could probably also report this state to me, but I haven't bothered using any yet.

**Additional context**

<details>
<summary>A screenshot of a successful `jj resolve --tool vscode` and a `jj st` that ran immediately after which reports "Conflict in parent commit has been resolved in working copy".
</summary>
<img width="563" alt="Screenshot 2024-12-24 at 1 28 35 PM" src="https://github.com/user-attachments/assets/39f5fec1-d2e6-4ba4-96da-d799c01bbaae" />
</details>

**Comment by arxanas** [comment]
If the working copy commit has conflicts (both generally and including the case of the `jj resolve` output), does it say `(conflict)` in the status line, like it does for the parent commit?

**Comment by davidbarsky** [comment]
Sorry for the delay. What do you mean by “status line”? Which command are you referring to?

**Comment by arxanas** [comment]
@davidbarsky specifically the "Working copy now at: " / "Working copy:" / "Parent commit:" lines in your screenshots.

**Comment by davidbarsky** [comment]
@arxanas Ah! Yes, I believe it does, if I'm understanding you correctly (see below). At the risk of re-emphasizing something already obvious to you, I think I'd "just" like `jj resolve` to affirmatively say "all conflicts are fixed" and not rely on me to infer that state through the absence additional warnings/details.

<img width="632" alt="Screenshot 2024-12-29 at 10 30 55 AM" src="https://github.com/user-attachments/assets/d55b018b-ec2d-4216-a4b6-1b4dd4fac946" />

**Comment by martinvonz** [comment]
Maybe #5111 will effectively address this?

**Comment by davidbarsky** [comment]
It would, yes! I'd also be happy to send a PR that copies [this snippet](https://github.com/jj-vcs/jj/blob/43ac4faac4780e45c55708dbaf55d5d95614e858/cli/src/commands/status.rs#L127-L135) from `jj status` over to `jj resolve`, if you'd like!

**Comment by arxanas** [comment]
> @arxanas Ah! Yes, I believe it does, if I'm understanding you correctly (see below). At the risk of emphasizing something already obvious to you, I'd "just" like `jj resolve` to affirmatively say "all conflicts are fixed" and not rely on me to infer that state through the absence of the normal conflict messages.

Thanks, I just didn't know about whether it showed the information that I thought it showed. (I think I typically don't use `jj resolve` and instead just open files in the editor and rely on the auto-resolving behavior.)

- Personally: Seeing the presence of `(conflict)` in the parent commit message and the absence of it in the working-copy commit is enough to rest assured that all conflicts have been resolved.
- Generally: It seems like it could be useful to print a summary of how many conflicts were resolved as a result of the `jj resolve` operation and how many remained as a result of the resolution, for which the transition from "conflicted parent commit" to "non-conflicted working-copy commit" would be a special case. This could give you a sense of how much progress you had made towards resolving conflicts.
- For now: I wouldn't be opposed to including the same `jj status` message you linked in the `jj resolve` output (either by copying it or abstracting it somehow).

Originally, I was just asking my question to understand the `jj resolve` UI and workflow better, and not because I had any specific recommendation about this issue. I'll defer to @martinvonz or others about the best design or implementation.

**Comment by martinvonz** [comment]
I agree with all @arxanas said. It might be best to wait for #5111 to land before adding the message to make sure it makes sense. Maybe it will mention the number of resolved conflicts too, as @arxanas said.

**Comment by davidbarsky** [comment]
Small update: #5111 helps, but I still find myself running `jj st` often to check for the blue "Hint: Conflict in parent commit has been resolved in working copy". Since writing this up, I've seen [some writing](https://blog.sshh.io/i/171208815/pattern-every-output-is-a-prompt) suggesting that it might be a good idea to provide "next steps" after CLI commands complete to guide AI tools.

1. AI tools may lack jj in their training data
2. Many users' first jj experience will be mediated by AI tools
    1. Anecdotally: A friend at Anthropic (historically, a self-described basic git user!) now runs multiple Claude Code instances in parallel and really struggles with git rebase and friends. I told that them jj would *literally* fix their pain points!

I suppose if I were to summarize, affordances for people (like myself...) whose attention spans might be, ah, lacking will help said people *and* machines in using jj (with a corresponding toggle to disable them globally, naturally).
